packages/snap/src/rpc/substrate/sendUnit.ts

```diff
@@ -2,10 +2,20 @@ import ApiPromise from "@polkadot/api/promise";
 import {Wallet} from "../../interfaces";
 import {Hash} from '@polkadot/types/interfaces/runtime';
 import {getKeyPair} from "../../polkadot/account";
+import {txEventEmitter} from "../../polkadot/events";
 
 export async function sendUnit(wallet: Wallet, api: ApiPromise, amount: string|number, to: string): Promise<Hash> {
   const keypair = await getKeyPair(wallet);
-  return await api.tx.balances
-    .transfer(to, amount)
-    .signAndSend(keypair);
+  const signedTx = api.tx.balances.transfer(to, amount).sign(keypair, {});
+  signedTx.send(({ status }) => {
+    const hexHash = status.hash.toHex();
+    if (status.isInBlock) {
+      txEventEmitter.emit("inBlock", hexHash, {});
+      txEventEmitter.removeAllListeners("inBlock", hexHash);
+    } else if (status.isFinalized) {
+      txEventEmitter.emit("finalized", hexHash, {});
+      txEventEmitter.removeAllListeners("finalized", hexHash);
+    }
+  });
```

Comment:
This is kinda wierd I must admin but I guess we can improve later. 
Think flow should be:
- sendUnit -> hash
- subscribeToStatus(hash).on("included", callback).on("finalized, callback)

Do you think this can work?
If yes, open issue and we will solve it after this milestone

Reply:
@mpetrunic I actually successfully changed subscription flow. The only thing that I am struggling to implement is to have this "chain" API. Currently flow is like this:

```
const status = subscribeToTxStatus(hash);
status.on("included", callback);
status.on("finalized", callback);
```

I tested this inside `index.html`, and it is working.